tests/rule_based_profiler/test_profiler_run_anonymizer.py

```diff
@@ -0,0 +1,525 @@
+                        "parent_class": "MetricMultiBatchParameterBuilder",
+                    }
+                ],
+            }
+        ],
+        "config_version": 1.0,
+        "rule_count": 1,
+        "variable_count": 1,
+    }
+
+
+def test_resolve_config_using_acceptable_arguments(
+    profiler_with_placeholder_args: RuleBasedProfiler,
+) -> None:
+    config: dict = RuleBasedProfilerConfig.resolve_config_using_acceptable_arguments(
+        profiler=profiler_with_placeholder_args
+    )
+
+    # Ensure we have expected keys while also removing unnecessary ones
+    assert all(
+        attr in config
+        for attr in ("name", "config_version", "rules", "variable_count", "rule_count")
+    )
+    assert all(
+        attr not in config for attr in ("class_name", "module_name", "variables")
+    )
+
+    assert config["variable_count"] == 1 and config["rule_count"] == 1
+
+
+def test_resolve_config_using_acceptable_arguments_with_runtime_overrides(
+    profiler_with_placeholder_args: RuleBasedProfiler,
+) -> None:
+    rule_name: str = "my_rule"
+    assert all(rule.name != rule_name for rule in profiler_with_placeholder_args.rules)
+
+    rules: Dict[str, dict] = {rule_name: {"foo": "bar"}}
+    config: dict = RuleBasedProfilerConfig.resolve_config_using_acceptable_arguments(
+        profiler=profiler_with_placeholder_args, rules=rules
+    )
+
+    assert len(config["rules"]) == 1 and rule_name in config["rules"]
+
+
+def test_resolve_config_using_acceptable_arguments_with_runtime_overrides_with_batch_requests(
+    profiler_with_placeholder_args: RuleBasedProfiler, usage_stats_profiler_config: dict
+) -> None:
+    datasource_name = "my_datasource"
+    data_connector_name = "my_basic_data_connector"
+    data_asset_name = "my_data_asset"
+
+    batch_request: BatchRequest = BatchRequest(
+        datasource_name=datasource_name,
+        data_connector_name=data_connector_name,
+        data_asset_name=data_asset_name,
+    )
+
+    # Add batch requests to fixture before running method
+    rules: Dict[str, dict] = usage_stats_profiler_config["rules"]
+    rules["rule_1"]["domain_builder"]["batch_request"] = batch_request
+
+    config: dict = RuleBasedProfilerConfig.resolve_config_using_acceptable_arguments(
+        profiler=profiler_with_placeholder_args, rules=rules
+    )
+
+    assert all(
+        attr in config
+        for attr in ("name", "config_version", "rules", "variable_count", "rule_count")
+    )
+
+    domain_builder: dict = config["rules"]["rule_1"]["domain_builder"]
+    converted_batch_request: dict = domain_builder["batch_request"]
+    assert converted_batch_request["datasource_name"] == datasource_name
+    assert converted_batch_request["data_connector_name"] == data_connector_name
+    assert converted_batch_request["data_asset_name"] == data_asset_name
+
+
+def test_all_builder_classes_are_recognized_as_profiler_run_anonymizer_attrs(
```

Comment:
❤️❤️❤️❤️❤️❤️❤️
Love this test. Until we can refactor this will help us not forget to add new builder classes to the anonymizer. Thank you for writing this, our future selves will also thank you.